src/test/scala/org/apache/spark/sql/delta/MergeIntoScalaSuite.scala

```diff
@@ -528,6 +534,161 @@ class MergeIntoScalaSuite extends MergeIntoSuiteBase  with DeltaSQLCommandTest {
       Nil)
   }
 
+  def withConcurrentCompactionTestSetup(toggle: Boolean)(closure: String => Unit): Unit = {
+    // in order to slow down the merge operation and increase the conflict probability,
+    // we use a UDF with sleep as the merge condition
+    spark.udf.register("slowMergeCondition", (condition: Boolean) => {
+      Thread.sleep(10)
+      condition
+    })
+
+    withTempDir { dir =>
+      withSQLConf(DeltaSQLConf.CONCURRENT_COMPACTION_CONFLICT_DETECTION.key -> toggle.toString,
+                  // we enable merge repartition before write to slow down the merge operation,
+                  // thus increasing conflict probability
+                  DeltaSQLConf.MERGE_REPARTITION_BEFORE_WRITE.key -> "true") {
+        closure(dir.getAbsolutePath)
+      }
+    }
+  }
+
+  val concurrentCompactionTestPartitions = 3
+  val concurrentCompactionTestRangeSize = 10
+  val concurrentCompactionTestNumIterations = 10
+
+  def createConcurrentCompactionTestTable(location: String): Unit = {
+    spark.range(0)
+            .withColumn("par", $"id" % concurrentCompactionTestPartitions)
+            .coalesce(1)
+            .write.partitionBy("par").format("delta").save(location)
+  }
+
+  def executeConcurrentCompactionTestInsertOnlyMerge(location: String, i: Int): Unit = {
+    val df = spark.range(i, i + concurrentCompactionTestRangeSize)
+            .withColumn("par", $"id" % concurrentCompactionTestPartitions)
+    val table = io.delta.tables.DeltaTable.forPath(spark, location)
```

Comment:
nit: add import

Reply:
I used the fully qualified name to be consistent with other tests in this suite, for example: `test("basic scala API")`.   
Should I keep it the same or change it?

Reply:
I'd at least do `import` just before the line. Just want to get rid of this FQCN here.

Reply:
I think I figured out why the other tests used the FQCN - the import order seems to have messed up the reference to `DeltaTable`.  
I believe I managed to fix that in 109cb55.